drivers/i2c/Kconfig.esp32

```diff
@@ -0,0 +1,102 @@
+# Kconfig.esp32 - ESP32 I2C configuration options
+#
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config I2C_ESP32
+	bool "ESP32 I2C"
+	depends on I2C && SOC_ESP32 && GPIO_ESP32
+	default n
+	help
+	  Enables the ESP32 I2C driver
+
+if I2C_ESP32
+
+config I2C_ESP32_TIMEOUT
+	int "I2C timeout to receive a data bit in APB clock cycles"
+	default 200000
+
+if I2C_0
+
+config I2C_0_DEFAULT_CFG
+	# Standard speed, master
+	default 0x12
+
+config I2C_ESP32_0_TX_LSB_FIRST
+	bool "Port 0 Transmit LSB first"
+	default n
+
+config I2C_ESP32_0_RX_LSB_FIRST
+	bool "Port 0 Receive LSB first"
+	default n
+
+config I2C_ESP32_0_IRQ
+	int "Port 0 IRQ line"
```

Comment:
Is this actually configurable?  Almost certain everything on an xtensa SoC is hardwired to specific IRQs.  Or maybe different ESP-32 variants use different lines?  If it can't change across hardware we actually know about, this should probably be a preprocessor symbol and not a kconfig.

Reply:
Yes, it's configurable. There are many IRQ lines that can be routed to peripherals; there's a call to a ROM function that controls the interrupt matrix to set this up.